pyomeca/rototrans.py

```diff
@@ -1,499 +1,273 @@
-            angles[2, :, :] = np.arctan2(self[1, 0, :], self[1, 1, :])
-        elif angle_sequence == "yzx":
-            angles[2, :, :] = np.arctan2(-self[1, 2, :], self[1, 1, :])
-            angles[0, :, :] = np.arctan2(-self[2, 0, :], self[0, 0, :])
-            angles[1, :, :] = np.arcsin(self[1, 0, :])
-        elif angle_sequence == "zxy":
-            angles[1, :, :] = np.arcsin(self[2, 1, :])
-            angles[2, :, :] = np.arctan2(-self[2, 0, :], self[2, 2, :])
-            angles[0, :, :] = np.arctan2(-self[0, 1, :], self[1, 1, :])
-        elif angle_sequence == "zyx":
-            angles[2, :, :] = np.arctan2(self[2, 1, :], self[2, 2, :])
-            angles[1, :, :] = np.arcsin(-self[2, 0, :])
-            angles[0, :, :] = np.arctan2(self[1, 0, :], self[0, 0, :])
-        elif angle_sequence == "zyz":
-            angles[0, :, :] = np.arctan2(self[1, 2, :], self[0, 2, :])
-            angles[1, :, :] = np.arccos(self[2, 2, :])
-            angles[2, :, :] = np.arctan2(self[2, 1, :], -self[2, 0, :])
-        elif angle_sequence == "zxz":
-            angles[0, :, :] = np.arctan2(self[0, 2, :], -self[1, 2, :])
-            angles[1, :, :] = np.arccos(self[2, 2, :])
-            angles[2, :, :] = np.arctan2(self[2, 0, :], self[2, 1, :])
-        elif angle_sequence == "zyzz":
-            angles[0, :, :] = np.arctan2(self[1, 2, :], self[0, 2, :])
-            angles[1, :, :] = np.arccos(self[2, 2, :])
-            angles[2, :, :] = np.arctan2(self[2, 1, :], -self[2, 0, :])
-
-        return angles
-
-    @staticmethod
-    def rt_from_euler_angles(
-        angles=FrameDependentNpArray(),
-        angle_sequence="",
-        translations=FrameDependentNpArray(),
-    ):
-        """
+            # if we provide data, we copy them to avoid making inplace changes
+            data = data.copy()
 
-        Parameters
-        ----------
-        angles : FrameDependentNpArray
-            Euler angles of the rototranslation
-        angle_sequence : str
-            Euler sequence of angles; valid values are all permutation of axes (e.g. "xyz", "yzx", ...)
-        translations : FrameDependentNpArray
-            Translation part of the Rototrans matrix
-
-        Returns
-        -------
-        rt : RotoTrans
-            The rototranslation associated to the input parameters
-        """
-        # Convert special zyzz angle sequence to zyz
-        if angle_sequence == "zyzz":
-            angles[2, :, :] -= angles[0, :, :]
-            angle_sequence = "zyz"
-
-        # If the user asked for a pure rotation
-        if angles.get_num_frames() != 0 and translations.get_num_frames() == 0:
-            translations = FrameDependentNpArray(
-                np.zeros((3, 1, angles.get_num_frames()))
-            )
+            if data.shape[0] != 4 or data.shape[1] != 4:
+                raise IndexError(
+                    f"data must have first and second dimensions of length 4, you have: {data.shape}"
+                )
```

Comment:
We could argue that a 3x3 should also be valid (Translation would be 0, 0, 0)

Reply:
Done in https://github.com/pyomeca/pyomeca/pull/88/commits/a59eb04eec21ed440a2b46f15c863341d393cf92